Q/QuEST/build_tarballs.jl

```diff
@@ -0,0 +1,52 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "QuEST"
+version = v"3.7.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/QuEST-Kit/QuEST.git", "d4f75f724993b4af8e43a796e3c09ce24ae11670")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/QuEST
+cmake \
+    -DCMAKE_C_STANDARD=99 \
+    -DCMAKE_INSTALL_PREFIX=$prefix \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_BUILD_TYPE=Release \
+    .
+make -j${nproc}
+mkdir -p "${includedir}"
+cp -vr $WORKSPACE/srcdir/QuEST/QuEST/include/* ${includedir}/
+install -Dvm 755 $WORKSPACE/srcdir/QuEST/libQuEST.${dlext} ${libdir}/libQuEST.${dlext}
+install_license LICENSE.txt
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("i686", "linux"; libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("i686", "linux"; libc = "musl"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("x86_64", "macos"; ),
+    Platform("aarch64", "macos"; )
+]
```

Comment:
Any reason why restricting to these platforms?
```suggestion
platforms = supported_platforms()
```

Reply:
1. Still very new, so my reasons are not as sound as they could be
2. When trying to get this to build locally, I kept getting an error then -mavx was not recognised (which I now know relates to builds that do are x86) the builds preset are the ones on the wizard run that did not exit completely.
3. On my local Ubuntu AMD I can compile this repo and run some arbitrary scripts when I remove all instances of AVX flags. I do not think it's even needed, but since I am not the owner, I am leaving it as is.
4. I would like this available to all platforms.